text-minimessage/src/main/java/net/kyori/adventure/text/minimessage/tag/resolver/CachingTagResolver.java

```diff
@@ -78,4 +79,21 @@ public void handle(final @NotNull Component serializable, final @NotNull ClaimCo
       ((SerializableResolver) this.resolver).handle(serializable, consumer);
     }
   }
+
+  @Override
+  public boolean equals(final @Nullable Object other) {
+    if (this == other) {
+      return true;
+    }
+    if (!(other instanceof CachingTagResolver)) {
+      return false;
+    }
+    final CachingTagResolver that = (CachingTagResolver) other;
+    return Objects.equals(this.cache, that.cache);
```

Comment:
Do we not need to check the resolver here as well?

Reply:
I don't think so, since it would be rare for someone to implement a TagResolver.WithoutArguments in a separate class (and even rarer to implement a proper equals method), everyone would use a lambda to implement it directly, making that between 2 identically created instances of CachingTagResolver and applying the same arguments to them, the only different thing is the resolver.
In case the TagResolver.WithoutArguments was a CachingTagResolver, in the only method where this class is created it will return the same resolver casted to a CachingTagResolver

Reply:
the cache should not be part of equality either way - it records thd output if thd underlying resolver, and two caching resolvers can draw from the same backing resolvers while having different sets of keys in their cache.

imo even if it's not always got an equals override, the backing resolver is the only sensible field to check for equality